test/functional/p2p_tx_download.py

```diff
@@ -216,6 +216,24 @@ def test_preferred_inv(self):
         with p2p_lock:
             assert_equal(peer.tx_getdata_count, 1)
 
+    def test_txid_inv_delay(self):
+        self.log.info('Check that inv from a txid-relay peers are delayed by {} s'.format(TXID_RELAY_DELAY))
+        self.restart_node(0, extra_args=['-whitelist=noban@127.0.0.1'])
+        mock_time = int(time.time() + 1)
+        self.nodes[0].setmocktime(mock_time)
+        peer = self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=False))
+        # Add a second wtxid-relay connection otherwise TXID_RELAY_DELAY is waived in
+        # lack of wtxid-relay peers
+        self.nodes[0].add_p2p_connection(TestP2PConn(wtxidrelay=True))
+        peer.send_message(msg_inv([CInv(t=MSG_TX, h=0xff11ff11)]))
+        peer.sync_with_ping()
+        with p2p_lock:
+            assert_equal(peer.tx_getdata_count, 0)
+        self.nodes[0].setmocktime(mock_time + TXID_RELAY_DELAY)
```

Comment:
I don't think you can achieve this with current p2p framework, we can write to the mocktime but can we read it from it ? A quick look, I don't find test doing it so not sure that's a feature of our current test framework. Note that was already the way done by legacy test. But lmk if there is way to do so I don't see

Reply:
> I don't find test doing it so not sure that's a feature of our current test framework

Yeah, I haven't seen an example either. The way I'd do it is just use `sleep()` instead of setmocktime... 2 seconds isn't too bad when the tests are running in parallel, but fosho not ideal.